### PR TITLE
refactor(text)!: fully remove spacing and text passage variants

### PR DESCRIPTION
**Pull request by jinlee93**
[EDS-837]
### Summary:
- with preflight, can now remove the `spacing` and `as="div"` text passage variants
- replaced the spacing provided with TW utility classes
  - text passage variant used `<LayoutLinelengthContainer>` which just applied max width of 36 rem, replaced with className `max-w-xl`
### Test Plan:
- compare the affected stories with the ones from https://chanzuckerberg.github.io/edu-design-system/
  - some small spacing differences are acceptable

### CODE REVIEW

**Comment by booc0mtaco** [comment]
@jinlee93 is merging this one blocked by #1490 ?

**Comment by jinlee93** [comment]
> @jinlee93 is merging this one blocked by #1490 ?

Yup, the PR target is that feature branch, so once that one lands, this one will auto-target `next`

**Comment by booc0mtaco** [comment]
> > @jinlee93 is merging this one blocked by #1490 ?
> 
> Yup, the PR target is that feature branch, so once that one lands, this one will auto-target `next`

Oh i see, thx

File: .storybook/pages/CoursePlannerEdit/CoursePlannerEdit.tsx
Comment (booc0mtaco) [inline .storybook/pages/CoursePlannerEdit/CoursePlannerEdit.tsx:272]: As part of the Cookbook RFC work, we will review such stories and make the text and such more generic (and use that FPO block)

File: src/components/FieldNote/FieldNote.stories.tsx
Comment (booc0mtaco) [inline src/components/FieldNote/FieldNote.stories.tsx:34]: I would have thought preflight would make resetting the bottom margin unnecessary? Unless this is doing something else...

File: src/components/Text/Text.module.css
Comment (jinlee93) [reply src/components/Text/Text.module.css:91]: Totally meant to remove that prop entirely, fixed

File: src/components/FieldNote/FieldNote.stories.tsx
Comment (jinlee93) [reply src/components/FieldNote/FieldNote.stories.tsx:34]: Yup, these classes are redundant, removed

File: .storybook/pages/CoursePlannerEdit/CoursePlannerEdit.tsx
Comment (jinlee93) [reply .storybook/pages/CoursePlannerEdit/CoursePlannerEdit.tsx:272]: Super down for that

File: src/components/Text/Text.module.css
Comment (jinlee93) [reply src/components/Text/Text.module.css:217]: Yup, doesn't need to be part of this text passage variant haha

File: src/components/Text/Text.tsx
Comment (jinlee93) [reply src/components/Text/Text.tsx:91]: What would that look like?

Comment (booc0mtaco) [reply src/components/Text/Text.tsx:85]: [Similar to this in `ClickableStyle`](https://github.com/chanzuckerberg/edu-design-system/blob/main/src/components/ClickableStyle/ClickableStyle.tsx#L103). Just a little syntax sugar.

Comment (jinlee93) [reply src/components/Text/Text.tsx]: Got it, changed

**Review by booc0mtaco — APPROVED** [review]
will review the combined ticket once this one is merged in to the preflight stuff

**Pull request merged**
